### ISSUE TITLE
java.lang.IllegalArgumentException

### ISSUE THREAD

**Issue by henrikra** [intercom/intercom-android#344]
Anyone else experiencing this crash? :O

## Version info
  - Intercom for Android version: 3.1.3
  - Android OS version: 4.4.2
  - Android Device: Samsung GT-N7100

## Expected behavior
Not to crash

## Actual behavior
Crash randomly

## Steps to reproduce
Random

## Logs
```
java.lang.IllegalArgumentException You must call this method on the main thread 
    Util.java:135 io.intercom.com.bumptech.glide.util.Util.assertMainThread
    GenericRequestBuilder.java:649 io.intercom.com.bumptech.glide.GenericRequestBuilder.into
    GenericTranscodeRequest.java:89 io.intercom.com.bumptech.glide.GenericTranscodeRequest.downloadOnly
    DrawableTypeRequest.java:96 io.intercom.com.bumptech.glide.DrawableTypeRequest.downloadOnly
    AvatarUtils.java:28 io.intercom.android.sdk.utilities.AvatarUtils.preloadAvatar
    OverlayManager.java:238 io.intercom.android.sdk.overlay.OverlayManager.preloadAvatarThenDisplayNotifications
    OverlayManager.java:106 io.intercom.android.sdk.overlay.OverlayManager.onStateChange
    OverlayManager.java:401 io.intercom.android.sdk.overlay.OverlayManager.refreshStateBecauseUserIdentityIsNotInStoreYet
    RealIntercom.java:196 io.intercom.android.sdk.RealIntercom.softRegister
    RealIntercom.java:149 io.intercom.android.sdk.RealIntercom.registerIdentifiedUser
    IntercomModule.java:47 com.robinpowered.react.Intercom.IntercomModule.registerIdentifiedUser
    Method.java:-2 java.lang.reflect.Method.invokeNative
    Method.java:515 java.lang.reflect.Method.invoke
    BaseJavaModule.java:368 com.facebook.react.bridge.BaseJavaModule$JavaMethod.invoke
    JavaModuleWrapper.java:138 com.facebook.react.cxxbridge.JavaModuleWrapper.invoke
    NativeRunnable.java:-2 com.facebook.react.bridge.queue.NativeRunnable.run
    Handler.java:733 android.os.Handler.handleCallback
    Handler.java:95 android.os.Handler.dispatchMessage
    MessageQueueThreadHandler.java:31 com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage
    Looper.java:146 android.os.Looper.loop
    MessageQueueThreadImpl.java:208 com.facebook.react.bridge.queue.MessageQueueThreadImpl$3.run
    Thread.java:841 java.lang.Thread.run
```

**Comment by ronocod** [comment]
Thanks for reporting this, it looks like this can happen when `registerIdentifiedUser()` is called from a background thread. We'll have a fix for this in our next release.

**Comment by samratshekhar** [comment]
+1
- Intercom SDK  version: 3.0.21
- Android OS: 6.0
- Device: Motorola Moto X Pure Edition XT1575

**Comment by henrikra** [comment]
@ronocod Good to hear! Looks like more people are having this crash too :)

**Comment by ronocod** [comment]
This is fixed in 3.2.2.

**Comment by henrikra** [comment]
We will install new version next week! I will inform if it happens again :)
